asteroid/models/base_models.py

```diff
@@ -222,9 +222,9 @@ def __init__(self, encoder, masker, decoder, encoder_activation=None):
         self.encoder = encoder
         self.masker = masker
         self.decoder = decoder
-
         self.encoder_activation = encoder_activation
         self.enc_activation = activations.get(encoder_activation or "linear")()
+        self.sample_rate = getattr(encoder, "sample_rate", None)
```

Comment:
Since the attribute is supposed to be read-only anyway, would having `sample_rate` as a property, like
```python
    @property
    def sample_rate(self):
        return get_attr(self.encoder, "sample_rate", None)
```
help in consistency?

Reply:
Thanks for the feedback. I don't think a property is needed here. You'd need a property if you expect the sample_rate to change in the encoder, so that it can be dynamically retrieved, right?

Reply:
True. Having a property which redirects to the value stored in the encoder allows to have one fewer attribute in `BaseEncoderMaskerDecoder`, ensures consistency between the value in `BaseEncoderMaskerDecoder` and the encoder and prevents silly mistakes like setting `sample_rate` on instances of `BaseEncoderMaskerDecoder` to a different value. I know, the latter case is not supposed to happen, but why take the risk?